src/httpd_jsonapi.c

```diff
@@ -2634,6 +2634,111 @@ jsonapi_reply_library_album_tracks(struct httpd_request *hreq)
   return HTTP_OK;
 }
 
+static int
+jsonapi_reply_library_tracks_get_byid(struct httpd_request *hreq)
+{
+  time_t db_update;
+  struct query_params query_params;
+  const char *track_id;
+  struct db_media_file_info dbmfi;
+  json_object *reply = NULL;
+  int ret = 0;
+
+  db_update = (time_t) db_admin_getint64(DB_ADMIN_DB_UPDATE);
+  if (db_update && httpd_request_not_modified_since(hreq->req, &db_update))
+    return HTTP_NOTMODIFIED;
+
+
+  track_id = hreq->uri_parsed->path_parts[3];
+
+  memset(&query_params, 0, sizeof(struct query_params));
+
+  query_params.type = Q_ITEMS;
+  query_params.sort = S_ALBUM;
+  query_params.filter = db_mprintf("(f.id = %q)", track_id);
+
+  ret = db_query_start(&query_params);
+  if (ret < 0)
+    goto error;
+
+  ret = db_query_fetch_file(&query_params, &dbmfi);
+  if (ret < 0)
+    goto error;
+
+  if (dbmfi.id == 0)
+    {
+      DPRINTF(E_LOG, L_WEB, "Track with id '%s' not found.\n", track_id);
+      ret = -1;
+      goto error;
+    }
+
+  reply = track_to_json(&dbmfi);
+
+  ret = evbuffer_add_printf(hreq->reply, "%s", json_object_to_json_string(reply));
+  if (ret < 0)
+    DPRINTF(E_LOG, L_WEB, "browse: Couldn't add track to response buffer.\n");
+
+ error:
+  db_query_end(&query_params);
+  free(query_params.filter);
+  jparse_free(reply);
+
+  if (ret < 0)
+    return HTTP_INTERNAL;
+
+  return HTTP_OK;
+}
+
+static int
+jsonapi_reply_library_tracks_put_byid(struct httpd_request *hreq)
+{
+  int track_id;
+  const char *param;
+  int val;
+  int ret;
+
+  ret = safe_atoi32(hreq->uri_parsed->path_parts[3], &track_id);
+  if (ret < 0)
+    return HTTP_INTERNAL;
+
+  // Update play_count/skip_count
+  param = evhttp_find_header(hreq->query, "play_count");
+  if (param)
+    {
+      if (strcmp(param, "increment") == 0)
+	{
+	  db_file_inc_playcount(track_id);
+	}
+      else if (strcmp(param, "reset") == 0)
+	{
+	  db_file_reset_playskip_count(track_id);
+	}
+      else
+	{
+	  DPRINTF(E_WARN, L_WEB, "Ignoring invalid play_count value '%s' for track '%d'.\n", param, track_id);
+	}
+    }
+
+  // Update rating
+  param = evhttp_find_header(hreq->query, "rating");
+  if (param)
+    {
+      ret = safe_atoi32(param, &val);
+      if (ret < 0)
+	return HTTP_BADREQUEST;
+
+      if (val >= 0 && val < DB_FILES_RATING_MAX)
```

Comment:
Is val == DB_FILES_RATING_MAX disallowed?

Reply:
Good catch! It is a valid value, i'll fix this.